pkg/advancedapi/advancedApi.go

```diff
@@ -310,83 +351,83 @@ func AddControlDelegationToDocument(resolverClient register.ResolverClient, name
 }
 
 // RemoveControlDelegationFromDocument Remove register control delegation proof from a register document.
-func RemoveControlDelegationFromDocument(resolverClient register.ResolverClient, name string, identity register.RegisteredIdentity) error {
-	return removeKeyFromDocument(resolverClient, name, identity)
+func RemoveControlDelegationFromDocument(resolverClient register.ResolverClient, originalDoc *register.RegisterDocument, name string, identity register.RegisteredIdentity) error {
+	return removeKeyFromDocument(resolverClient, originalDoc, name, identity)
 }
 
 // RevokeControlDelegationFromDocument Set register control delegation proof revoke field.
-func RevokeControlDelegationFromDocument(resolverClient register.ResolverClient, name string, identity register.RegisteredIdentity) error {
-	return revokeKeyFromDocument(resolverClient, name, identity)
+func RevokeControlDelegationFromDocument(resolverClient register.ResolverClient, originalDoc *register.RegisterDocument, name string, identity register.RegisteredIdentity) error {
+	return revokeKeyFromDocument(resolverClient, originalDoc, name, identity)
 }
 
-// DelegateAuthentication Delegate authentication between delegating registered identity and delegated registered identity.
-func DelegateAuthentication(resolverClient register.ResolverClient, delegatingKeyPair *crypto.KeyPair, delegatingDid string, subjectKeyPair *crypto.KeyPair, subjectDid string, name string) error {
-	delegatingDoc, err := GetRegisterDocument(resolverClient, delegatingDid)
-	if err != nil {
-		return err
+// DelegationOpts Options for delegation
+type DelegationOpts struct {
+	ResolverClient     register.ResolverClient
+	DelegatingKeyPair  *crypto.KeyPair
+	DelegatingDid      string
+	DelegatingDocument *register.RegisterDocument
```

Comment:
Based on [this](https://github.com/uber-go/guide/blob/master/style.md#omit-zero-value-fields-in-structs), maybe it makes more sense to leave out explicit `nil` for `DelegatingDocument` & `SubjectDocument` where it's not used.

Reply:
I've left the nils in the main_test but removed them from internal uses like regular api